### PR TITLE
Don't show deactivated tracks in embed player

### PR DESCRIPTION
**Pull request by rickyrombo**
Also:
- Don't show deleted tracks in collections embed player
- Add a bit of typing:
  - `Track` (for `/v1/tracks` API), `FullTrack` for `v1/full/tracks` API, and `TrackModel` (for `/tracks` API)
  - `User`, `FullUser`, and `UserModel` (respecting the same naming convention above)

### CODE REVIEW

File: src/types/Track.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  track_segments: TrackSegment[]
+  updated_at: string
+  user_id: string
+  user: FullUser
+  is_delete: boolean
+  cover_art: string
+  remix_of: FullRemixParent
+}
+
+export type TrackModel = {
```
Comment (rickyrombo) [inline src/types/Track.ts:103]: We have three different models for tracks that can be returned from the API:
1) what is returned from `v1/tracks`
2) what is returned from `v1/full/tracks`
3) what is return from `/tracks`

Dream: consolidate these into just the first two, automate typing in TS by generating from the models.py on discovery, publish types with libs.

Comment (sliptype) [reply src/types/Track.ts:103]: 100% agree, I would love to have the types be automated

File: src/servlets/bedtime/index.ts [hunk 2]
```diff
@@ -51,37 +51,39 @@ const getTrackMetadata = async (trackId: number, ownerId: number | null): Promis
 const getTracksFromCollection = async (collection: any, ownerUser: any): Promise<TrackResponse[]> => {
 
   const trackIds: number[] = collection.playlist_contents.track_ids.map((t: {time: number, track: number }) => t.track)
-  let tracks = []
+  let tracks: TrackModel[] = []
 
   // Fetch tracks if there are IDs
   if (trackIds.length) {
     const unordredTracks = await getTracks(trackIds)
```
Comment (sliptype) [inline src/servlets/bedtime/index.ts:58]: Just noticed this typo here

```suggestion
    const unorderedTracks = await getTracks(trackIds)
```

**Review by sliptype — APPROVED** [review]
Nice work!

File: src/types/Track.ts (hunk 1 above)
Comment (raymondjacobson) [reply src/types/Track.ts:103]: 👍 let's do this work in libs

File: src/types/Track.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  credits_splits: string
+  remix_of: any
+  create_date: string
+  release_date: string
+  file_type: string
+  description: string
+  license: string
+  isrc: string
+  iswc: string
+  track_segments: any
```
Comment (raymondjacobson) [inline src/types/Track.ts:128]: worth biting the bullet now and making sure these are "Correct" too? https://github.com/AudiusProject/audius-client/blob/master/src/common/models/Track.ts#L77

**Review by raymondjacobson — APPROVED** [review]

**Comment by raymondjacobson** [comment]
doing gods work

**Review by raymondjacobson — APPROVED** [review]

**Review by sliptype — APPROVED** [review]

**Pull request merged**